Riot/Modules/Settings/Security/SecurityViewController.m

```diff
@@ -524,12 +559,55 @@ - (void)setUpcrossSigningButtonCellForBootstrap:(MXKTableViewCellWithButton*)but
     [buttonCell.mxkButton setTitle:btnTitle forState:UIControlStateNormal];
     [buttonCell.mxkButton setTitle:btnTitle forState:UIControlStateHighlighted];
  
-    [buttonCell.mxkButton addTarget:self action:@selector(bootstrapCrossSigning:) forControlEvents:UIControlEventTouchUpInside];
+    [buttonCell.mxkButton addTarget:self action:@selector(setupCrossSigning:) forControlEvents:UIControlEventTouchUpInside];
 }
 
-- (void)bootstrapCrossSigning:(UITapGestureRecognizer *)recognizer
+- (void)setupCrossSigning:(UITapGestureRecognizer *)recognizer
 {
+#ifdef NEW_CROSS_SIGNING_FLOW
+    // TODO: Implement the true setup flow
+    MXCrossSigning *crossSigning =  self.mainSession.crypto.crossSigning;
+    if (crossSigning)
+    {
+        [self startActivityIndicator];
+        [crossSigning bootstrapWithPassword:@"password" success:^{
+            [self stopActivityIndicator];
+            [self reloadData];
+        } failure:^(NSError * _Nonnull error) {
+            [self stopActivityIndicator];
+            [self reloadData];
+            
+            [[AppDelegate theDelegate] showErrorAsAlert:error];
+        }];
+    }
+#else
     [self displayComingSoon];
+#endif
+}
+
+- (void)resetCrossSigning:(UITapGestureRecognizer *)recognizer
+{
+    [currentAlert dismissViewControllerAnimated:NO completion:nil];
+    
+    // Double confirmation
+    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Are you sure?"  // TODO
+                                                                             message:@"You will need to verify trusted users again. Users who trusted you will need to verify you again."     // TODO
+                                                                      preferredStyle:UIAlertControllerStyleAlert];
+    
+    [alertController addAction:[UIAlertAction actionWithTitle:@"Reset"
+                                                        style:UIAlertActionStyleDefault
+                                                      handler:^(UIAlertAction * action)
+                                {
+                                    // Setup and reset are the same thing
+                                    [self setupCrossSigning:recognizer];
```

Comment:
Weird to send the "reset" button as parameter of the "setupCrossSigning:" action. We can use nil in this case.
```suggestion
                                    [self setupCrossSigning:nil];
```